src/main/java/seedu/address/model/transaction/Value.java

```diff
@@ -0,0 +1,56 @@
+package seedu.address.model.transaction;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a transaction's value.
+ */
+public class Value {
+    public static final String MESSAGE_CONSTRAINTS =
+            "Transaction value should only contain numbers and at most one dot, and it should not be blank";
+
+    public static final String VALIDATION_REGEX = "-?(?:\\d+(?:\\.\\d+)?|\\.\\d+)";
+    public final String value;
```

Comment:
Might want to consider using BigDecimal to store the value in future implementations, so that we can conduct operations such as summing, have a consistent representation of money, and also guard against potential bugs.

Reply:
I have thought about this and prefer to keep value as String for consistency with attributes of other classes. Change it to BigDecimal will result modifications in parsers.

For summing we can create BigDecimal from the String value, then convert back to String and create new editedTxn object.